deploy-basic-service/index.js

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env node
+
+const {
+	getInput,
+	debug, info,
+	startGroup, endGroup,
+	setFailed,
+} = require('@actions/core')
+const last = require('lodash/last')
+const {exec} = require('@actions/exec')
+const glob = require('@actions/glob')
+const {readFileSync, writeFileSync} = require('fs')
+const {join: pJoin} = require('path')
+
+const {
+	GITHUB_REPOSITORY,
+	GITHUB_ACTOR,
+	GITHUB_REF,
+	GITHUB_SHA,
+	GITHUB_WORKSPACE,
+} = process.env
+const [GITHUB_ORG, GITHUB_REPO] = GITHUB_REPOSITORY.split('/')
+const COMMIT = GITHUB_SHA.slice(0, 7)
+const DIR = GITHUB_WORKSPACE
+
+const buildDockerImage = async (dockerRepo, dockerTag) => {
+	const tag = 'publictransport/' + dockerRepo + ':' + dockerTag
+	await exec('docker', ['build', '-t', tag, '--iidfile', '/tmp/iid.txt', '.'])
+	const hash = await readFileSync('/tmp/iid.txt', { encoding: 'utf8' })
+
+	return {tag, hash}
+}
+
+const pushImageToHub = async (user, token, tag, hash) => {
+	await exec('docker', ['login', '-u', user, '--password-stdin'], {
+		input: token,
+	})
+	await exec('docker', ['push', tag])
+
+	let rawId = ''
+	await exec('docker', ['image', 'inspect', hash, '-f', '{{index .RepoDigests 0}}'], {
+		listeners: {
+			stdout: data => { rawId += (data.toString()) }
+		}
+	})
+	const id = rawId.replace(/\s/gi, '')
+	info('Docker image ID: ' + id) // todo: debug-level
+
+	return id
+}
+
+const deployToK8s = async (kubeconfig, imageId) => {
+	const pKubecfg = pJoin(DIR, '.kubeconfig')
+	await writeFileSync(pKubecfg, kubeconfig)
+
+	const deploymentResources = []
+	const globber = await glob.create(pJoin(DIR, 'kubernetes/**/*.yaml'))
+	for await (const resource of globber.globGenerator()) {
+		// replace image placeholder
+		debug(`replacing image placeholder for "${resource}"`)
+		await exec('sed', ['-e', `s|<IMAGE>|${imageId}|`, '-i', resource])
```

Comment:
Mutating Git-tracked files in an actions seems dangerous. Subsequent `git` commands or additional actions might complain or even do unexpected things. Because `kubectl` is able to take the definitions via `stdin`, let's use that.